### ISSUE TITLE
Allow customization of vendor dir

### ISSUE THREAD

**Issue by faizanakram99** [brefphp/bref#226]
I get this error when I try local deployment of a Symfony app with bref.

> Warning: require(/var/task/vendor/autoload.php): failed to open stream: No such file or directory in /opt/bootstrap on line 12
> PHP Fatal error:  require(): Failed opening required '/var/task/vendor/autoload.php' (include_path='.:/opt/bref/lib/php') in /opt/bootstrap on line 12

As such I can't even bootstrap my app

This error is due to the fact it tries to search for vendor directory in RootDir/vendor, in my app the vendor dir is located in other location RootDir/symfony/vendor and it is specified properly in composer.json

vendor dir is customizable (https://getcomposer.org/doc/06-config.md#vendor-dir) and bref should also fetch vendor dir from composer.json file (or environment variable)

**Comment by roenschg** [comment]
I have the same problem and what you proposed @atrope unfortunatly does not help in this case. 

The problem lies in the bootstrap file of bref in the following line: https://github.com/brefphp/bref/blob/master/runtime/layers/fpm/bootstrap#L12

It tries to include the autoload.php in the root directory.

If I have the following structure in my project it will work fine:
`
composer.json
vendor/
  ...
  autoload.php
src/
  lambda1/
    index.php
  lambda2/
    index.php
`

But if I have a structure like the following:
`
src/
  lambda1/
    composer.json
    vendor/
      ...
      autoload.php
    index.php
  lambda2/
    composer.json
    vendor/
      ...
      autoload.php
    index.php
`

It will fail because it cannot find the autoload.php file.

I see the second code structure as a common case if my two lambdas are using different dependencies and I want every lambda to be as small as possible.

**Comment by jenschude** [comment]
What you always can come up with is a modified bootstrap file in the root of your project.

**Comment by mnapoli** [comment]
This has been fixed in [v0.5.15](https://github.com/brefphp/bref/releases/tag/0.5.15).

**Comment by sc0ttdav3y** [comment]
Although this works, there's no documentation I could find that describes how to use it, so it took me about a day to understand how to get it going in the real world.  Here's what worked for me in case it helps others.

### 1. Folder structure

My folder structure includes a `services/` directory with a heterogeneous list of lambda functions using JS and PHP. 

```
services/
  test-php-api/
    handler.php
    composer.json
    lib/
      ...
    vendor/
      autoload.php
      ...
serverless.yml
```

### 2. serverless.yml

In `serverless.yml` I have the following:

```
functions:
  test-php-api:
    runtime: provided.al2
    handler: ./services/test-php-api/api-test-string.php
    layers:
      - ${bref:layer.php-74}
    events:
      - http:
          path: php-test-api
          method: get
          cors: true
    environment:
      BREF_AUTOLOAD_PATH: ./services/test-php-api/vendor/autoload.php
```

**Comment by deleugpn** [comment]
Thanks for adding your solution @sc0ttdav3y. During your development, did you by any chance run across this part of the documentation? https://bref.sh/docs/environment/php.html#custom-vendor-path

**Comment by sc0ttdav3y** [comment]
Hi @deleugpn, I don't think I saw that at the time — thanks for pointing that out 👍🏻 

My use-case is to set up **multiple separate PHP lambda functions in a nested directory structure**, and reading that doc  now, it doesn't explain how to do that (which I imagine would be a very common reason for wanting to customise the vendor folder location).

This nested directory structure seems quite common in serverless-land for JS lambdas, and I imagine it's the way anything other than a micro project would be configured. But it took me a while to work out how to have separate PHP code (and thus separate vendor folder) per lambda function for bref. In the end, my a-ha moment came when I realised I can put `environment.BREF_AUTOLOAD_PATH` into `functions.XXX` so it can point to different vendor paths _per lambda function_ (not the `provider` section as per the docs). 

I didn't mention it previously, but the other a-ha moment I had that I didn't write about here was using `package.patterns` at the top level with `individually: true` and then control which files are included in each lambda function, to split each lambda and have them only contain their own code.

This info is probably a serverless-101 thing, not a Bref thing, but I didn't know it and I needed to do it with Bref, which is why I added the worked example for others. 

Side note: overall I've found the Bref docs to be excellent.
